### ISSUE TITLE
Sanity check for applicant email format

### ISSUE THREAD

**Issue by dergigi** [OpenSats/website#77]
It seems like anything goes in the email field as of now.

Would be good to make sure that a simple typo doesn't remove what might be our online line of communication for some applicants.

**Comment by Bayernatoor** [comment]
Couple ideas for the above. 

The website uses Sendgrid to manage emails, they offer [email verification](https://sendgrid.com/solutions/email-api/email-address-validation-api/), however, it's only available to pro/premier plans and they offer 2500/5000 verifications per month. It works pretty well and offers email suggestions if you happen to make a typo. That being said I don't know which sendgrid plan is being used atm.

Another option is [Reacher](https://github.com/reacherhq/check-if-email-exists/tree/master), which is Open source and has a [paid version](https://reacher.email/pricing) or a self-hosted version. I have used it in the past and it's pretty reliable.

**Comment by abitcoinperson** [comment]
can't tell if this concern is more about deliverability (aka using a verfiication tool might makes sense) or if its just about syntax (ie make sure an email is formatted correctly)

**Comment by Bayernatoor** [comment]
Ya I was wondering the same thing but after testing the field it seems to check for proper syntax but not whether the email is actually valid. i didn't look further into the code to see how we're handling that though.

**Comment by abitcoinperson** [comment]
maybe an easy fix is:

Enter Email : ____________
Confirm Email: ____________

**Comment by Bayernatoor** [comment]
> maybe an easy fix is:
> 
> Enter Email : ____________ Confirm Email: ____________
> 
> probably better than pinging some third party with the emails of a bunch of FOSS devs

hehe absolutely and much easier to implement.

**Comment by dergigi** [comment]
Apologies for not being clear enough in the issue description.

The important bit is to have a *sanity check*, any sanity check, that makes sure that the text string that is entered is an email address. It's not about whether the email address can actually receive mails or is correct, it's about catching typos like `email@foo,com` - there should be a period `.com` but there is a comma `,com`.

<img width="192" alt="Screenshot 2023-11-17 at 09 24 48" src="https://github.com/OpenSats/website/assets/109058/7f3b6921-a727-4c09-abda-20dc91e9c5ab">

The double-email fields are something I hate, as I assume most people copy-paste whatever is in the first box. Leading to this:

<img width="270" alt="Screenshot 2023-11-17 at 09 25 15" src="https://github.com/OpenSats/website/assets/109058/f17fc8de-e350-45eb-8c4d-d0ead230750d">

So what I was trying to say in the issue description is that we should have a simple email format check, leading to something like this:

<img width="318" alt="Screenshot 2023-11-17 at 09 27 17" src="https://github.com/OpenSats/website/assets/109058/6b23477c-4065-4652-8641-ff76a7e98cb7">

We can use the nodejs [email-validator](https://www.npmjs.com/package/email-validator) which makes validation as easy as this:

```
var validator = require("email-validator");
validator.validate("test@email.com");
```

(Please don't use custom regex or anything like that.)

**Comment by Bayernatoor** [comment]
Yep this makes sense. I'll add this. 

(Please don't use custom regex or anything like that.)

Absolutely not.

**Comment by Bayernatoor** [comment]
Okay after poking around it seems that what @moteanup24 mentioned in #178 aligns with what you said above. Unless I am totally missing the boat here and apologies if I am but email syntax is being checked. It won't catch things like misspelled domains or TLDs but it'll catch invalid email syntax.

![image](https://github.com/OpenSats/website/assets/55212954/cb44b247-a7c5-4362-85e9-682dfefae65c)

**Comment by moteanup24** [comment]
Thanks @Bayernatoor 

FYI I've gone ahead and incorporated email-validator as @dergigi suggested (see #182). It allows for more explicit messaging to the user (error text in red) and seems to be more comprehensive. Just wanted to avoid any duplication of effort :)

**Comment by Bayernatoor** [comment]
closing this PR addressed by #182
